### PR TITLE
Update mimir-prometheus to 8a7ef4cfac

### PR DESCRIPTION
**Pull request by colega**
#### What this PR does

Updates mimir-prometheus to the latest commit.

This includes, among others, changes from https://github.com/grafana/mimir-prometheus/pull/462 that should fix https://github.com/grafana/mimir/issues/4564

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/mimir/issues/4564

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: vendor/github.com/prometheus/prometheus/model/labels/labels_string.go [hunk 1]
```diff
@@ -599,9 +599,12 @@ func (b *Builder) Get(n string) string {
 }
 
 // Range calls f on each label in the Builder.
-// If f calls Set or Del on b then this may affect what callbacks subsequently happen.
 func (b *Builder) Range(f func(l Label)) {
-	origAdd, origDel := b.add, b.del
+	// Stack-based arrays to avoid heap allocation in most cases.
+	var addStack [1024]Label
```
Comment (colega) [inline vendor/github.com/prometheus/prometheus/model/labels/labels_string.go:604]: I see that @bboreham has [reduced this to 128](https://github.com/prometheus/prometheus/pull/12174/files), I wonder if it's a big cost here, or we can go with 1024 and then bring the next commit?

**Comment by colega** [comment]
@krajorama this seems to bring some support for float histograms, would you mind to take a look?

File: vendor/github.com/prometheus/prometheus/storage/remote/codec.go [hunk 2]
```diff
@@ -120,30 +120,46 @@ func ToQueryResult(ss storage.SeriesSet, sampleLimit int) (*prompb.QueryResult,
 	for ss.Next() {
 		series := ss.At()
 		iter = series.Iterator(iter)
-		samples := []prompb.Sample{}
```
Comment (krajorama) [inline vendor/github.com/prometheus/prometheus/storage/remote/codec.go:123]: note only: adds support for histogram chunks in remote read, ok

**Review by krajorama — APPROVED** [review]
Native histogram related things look ok. Small fixes and a feature.
The EncOOOXOR replaced with a bool is removing a weirdness, which is good. But adds a new state which can always lead to unexpected results - otoh change seems straightforward.

**Comment by colega** [comment]
Thank you very much for your review, @krajorama <3

**Pull request merged**